core/src/test/java/org/apache/iceberg/jdbc/TestJdbcCatalog.java

```diff
@@ -630,6 +630,30 @@ public void testCreateNamespaceWithMetadata() {
     Assert.assertTrue(catalog.namespaceExists(testNamespace));
   }
 
+  @Test
+  public void testNamespaceLocation() {
+    Namespace testNamespace = Namespace.of("testDb", "ns1", "ns2");
+
+    // Test with location
+    Map<String, String> testMetadata = ImmutableMap.of();
+    catalog.createNamespace(testNamespace, testMetadata);
+
+    Assert.assertTrue(catalog.loadNamespaceMetadata(testNamespace).containsKey("location"));
+  }
+
+  @Test
+  public void testNamespaceCustomLocation() {
+    Namespace testNamespace = Namespace.of("testDb", "ns1", "ns2");
+    String namespaceLocation = "file:///tmp/warehouse/ns/path";
+
+    // Test with location
+    Map<String, String> testMetadata = ImmutableMap.of("location", namespaceLocation);
+    catalog.createNamespace(testNamespace, testMetadata);
+
+    Assert.assertTrue(
+        catalog.loadNamespaceMetadata(testNamespace).get("location").equals(namespaceLocation));
```

Comment:
nit: `Assertions.assertThat(catalog.loadNamespaceMetadata(testNamespace).get("location")).isEqualTo(namespaceLocation` might be preferrable here